pulser-core/pulser/backend/remote.py

```diff
@@ -161,3 +173,34 @@ def _type_check_job_params(job_params: list[JobParams] | None) -> None:
                     "All elements of 'job_params' must be dictionaries; "
                     f"got {type(d)} instead."
                 )
+
+    def open_batch(self) -> Self:
+        """Create an open batch that can continue recieve new job.
+
+        As long as jobs are submitted within an open context
+        manager.
+
+        Returns:
+            A class instance with an associated batch_id property
+        """
+        # Create batch and receive submission id
+        submission = cast(
+            RemoteResults, self._connection.submit(self._sequence, open=True)
+        )
+        self.batch_id = submission._submission_id
+        return self
+
+    def __enter__(self) -> Self:
+        # enter returns an instance of self to use open_batch
+        # as a context manager
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
+        # On context exit, we make a remote call to close the open batch
+        self._connection._close_batch(self.batch_id)
+        self.batch_id = ""
```

Comment:
Here's my issue with defining these methods here: 
```
backend = QPUBackend(...)
with backend:
   backend.run(...)  # Fails because the batch was not opened and there's no batch ID
``` 
or also
```
backend.open_batch() # This still opens the batch
backend.run(...)  # A batch_id is stored so this will add to the open batch
```

This is why I suggested that, instead of returning `Self`, `open_batch()` could return an auxiliary class specifically designed to handle open batches. Also, I guess the batch should
only be opened on `__enter__()`.

Reply:
I am not sure I have a mental image of what you are expecting when you are thinking we return a class these are bound to. Unless you share provide a simple example?

I wrote these in an inherited parent class initially which I kept getting a method order problem for, so I just removed the inherited class and brought them one step lower.

Reply:
Alright, here's an example:

```
class _OpenBatchManager:
    def __init__(self, backend: RemoteBackend):
        self.backend = RemoteBackend
    def __enter__(self) -> _OpenBatchManager:
        self.backend._open_batch = True
    def __exit__(self, ...) -> None:
        self.backend._open_batch = False
        self.backend._batch_id = ""
```

Then, in `RemoteBackend` you'll have:
```
   def open_batch(self) -> _OpenBatchManager:
       return _OpenBatchManager(self)
    
    def run(
        self, job_params: list[JobParams] | None = None, wait: bool = False
    ) -> RemoteResults:
        results = self._connection.submit(
            self._sequence, job_params=job_params, wait=wait, open=self._open_batch, batch_id=self._batch_id
        )
        if self._open_batch:
            self._batch_id = results._submission_id
            self._open_batch = False
        return results
```